internal/storage/writer/talaria/talaria.go

```diff
@@ -0,0 +1,62 @@
+package talaria
+
+import (
+	"context"
+	"sync"
+
+	talaria "github.com/kelindar/talaria/client/golang"
+	"github.com/kelindar/talaria/internal/encoding/key"
+	"github.com/kelindar/talaria/internal/monitor/errors"
+)
+
+// Writer to write to TalariaDB
+type Writer struct {
+	lock     sync.Mutex
+	endpoint string
+	client   *talaria.Client
+}
+
+// New initializes a new Talaria writer.
+func New(endpoint string) (*Writer, error) {
+	talaria, err := talaria.Dial(endpoint)
```

Comment:
Also, Dial can take more inputs
For example
`client.WithCircuit(time.Duration(grpcConfig.Timeout)*time.Second, grpcConfig.MaxConcurrent, grpcConfig.ErrorThreshold),
			client.WithNetwork(time.Duration(grpcConfig.DialTimeout)*time.Second),
			client.WithNonBlock()`

Refer https://github.com/kelindar/talaria/blob/master/client/golang/client_opts.go

Reply:
Thanks! Will take a look, completely missed out on this